command-expansion-server/src/worker.ts

```diff
@@ -81,6 +82,12 @@ export async function executeExpansion(opts: {
         commands = [commands];
       }
       const commandsFlat = commands.flat() as Command[];
+      for (const command of commandsFlat) {
+        (command as Mutable<Command>).metadata  = {
```

Comment:
I trust that you're doing the right thing here but I'm not sure I follow what's going on with `Mutable<>`.

Would a more functional map-style approach not work? I'm assuming you want these to be mutable as an optimization?

Reply:
Going to resolve this conversation and merge this PR as I am blocked. Dylan still seems to be out today, but he might see this and respond to your question. I don't have an answer for you :(